### PR TITLE
Adding test for bridge in water case

### PR DESCRIPTION
**Pull request by MikeGost**
Two changes here:

1. Adding a test case that maintains parity with the previous slicing logic for bridges crossing water.
2. Updating the TestAtlasHandler to refer loading from a generic osm resource. The boolean flag will identify whether it's a JOSM-based file or not, the method name didn't make sense before.

### CODE REVIEW

File: src/integrationTest/java/org/openstreetmap/atlas/geography/atlas/raw/OsmPbfToSlicedRawAtlasTest.java [hunk 1]
```diff
@@ -99,4 +104,31 @@ public void testPbfToSlicedRawAtlas()
...
+        final Atlas rawAtlas = rawAtlasGenerator.build();
+
+        final Set<IsoCountry> countries = new HashSet<>();
+        countries.add(IsoCountry.forCountryCode("CUB").get());
+        final RawAtlasCountrySlicer slicer = new RawAtlasCountrySlicer(countries, map, loadingArea);
+
+        final Atlas slicedAtlas = slicer.slice(rawAtlas);
+
+        // Make sure that the big bridge over water made it to the Atlas
+        Assert.assertNotNull(slicedAtlas.line(308541861000000L));
```
Comment (matthieun) [inline src/integrationTest/java/org/openstreetmap/atlas/geography/atlas/raw/OsmPbfToSlicedRawAtlasTest.java]: I believe that way is already inside the CUB boundary supplied for the test case, which will succeed, no matter if the code handles outside ways or not.

**Comment by MikeGost** [comment]
On second thought, it makes more sense to isolate this to a single test case that will test various boundary scenarios. Closing for now and reopening when updated.
